### PR TITLE
[GH-#205] Moved validation for tx types.

### LINKED ISSUE #205: move validation for tx types in tx structure as functions
in SignedTx add function `validate(tx)` which matches the type of `tx.data` included and calls the specific validate function e.g. `TxData.validate(tx.data)`

### PR DESCRIPTION
**Pull request by DanielaIvanova**
Closes #205

### CODE REVIEW

File: apps/aecore/lib/aecore/structures/signed_tx.ex [hunk 1]
```diff
@@ -29,9 +29,11 @@ defmodule Aecore.Structures.SignedTx do
     tx.data.from_acc == nil && tx.signature == nil
   end
 
-  @spec is_valid?(SignedTx.t()) :: boolean()
-  def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+  @spec validate(SignedTx.t()) :: boolean()
+  def validate(tx) do
+    case tx.data do
+      %SpendTx{} -> SpendTx.validate(tx)
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/structures/signed_tx.ex]: pass `tx.data` and validate signature seperately

File: apps/aecore/lib/aecore/structures/signed_tx.ex [hunk 2]
```diff
@@ -32,7 +32,7 @@ defmodule Aecore.Structures.SignedTx do
   @spec validate(SignedTx.t()) :: boolean()
   def validate(tx) do
     case tx.data do
-      %SpendTx{} -> SpendTx.validate(tx)
+      %SpendTx{} -> SpendTx.validate(tx.data) && Keys.verify_tx(tx)
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/structures/signed_tx.ex]: good for now, with more transactions we should move it to after the `case ... end`

**Comment by thepiwo** [comment]
@DanielaIvanova can you resolve the conflicts with the changes to master

**Comment by cytadela8** [comment]
Looks good, but I also liked `is_valid?` more.

**Pull request merged**